### PR TITLE
@preconcurrency import downgrades globals concurrency errors to warnings

### PR DESCRIPTION
**Pull request by sophiapoirier**

### CODE REVIEW

**Comment by sophiapoirier** [comment]
@swift-ci please test

**Comment by sophiapoirier** [comment]
@swift-ci please test Linux

File: lib/Sema/TypeCheckConcurrency.cpp [hunk 1]
```diff
@@ -4436,6 +4436,8 @@ ActorIsolation ActorIsolationRequest::evaluate(
         (isolation != ActorIsolation::NonisolatedUnsafe)) {
       auto *classDecl = var->getDeclContext()->getSelfClassDecl();
       const bool isActorType = classDecl && classDecl->isAnyActor();
+      const bool isPreconcurrencyUnspecifiedIsolation =
+          var->preconcurrency() && isolation.isUnspecified();
```
Comment (sophiapoirier) [inline lib/Sema/TypeCheckConcurrency.cpp:4440]: I am interested in feedback as to whether here there is any meaningful difference between testing for `var->preconcurrency()` vs `isolation.preconcurrency()`?

Comment (xedin) [reply lib/Sema/TypeCheckConcurrency.cpp:4440]: If I'm reading everything correctly ActorIsolation produced by `getIsolationFromAttributes` would not have "preconcurrency" bit set for explicitly nonisolated or unspecified. It seems better to use `var->preconcurrency()` just like you did in this case.

**Comment by sophiapoirier** [comment]
@swift-ci please test

File: lib/Sema/TypeCheckConcurrency.cpp (hunk 1 above)
Comment (sophiapoirier) [reply lib/Sema/TypeCheckConcurrency.cpp]: Thank you. It turned out that my logic was more deeply flawed in that those are checks for `@preconcurrency` as an attribute on a declaration, but do not capture when `@preconcurrency import` is used. I just pushed a replacement commit that undoes most of the previous one and appropriately manages diagnostic level changes based on `@preconcurrency import` usage instead.

**Comment by sophiapoirier** [comment]
@swift-ci please test

**Comment by sophiapoirier** [comment]
@swift-ci please test

**Comment by sophiapoirier** [comment]
@swift-ci please test Windows

**Review by hborla — APPROVED** [review]

File: lib/Sema/TypeCheckConcurrency.cpp (hunk 1 above)
Comment (hborla) [reply lib/Sema/TypeCheckConcurrency.cpp]: I've been bumping into issues with `Decl::preconcurrency` lately. It does not include `@preconcurrency` attributes that are inferred. It basically represents explicit `@preconcurrency` annotations + it's always true for imported declarations from ObjC.

If there are cases where `ActorIsolation` does not preserve the preconcurrency bit, we should fix that.

**Review by xedin — APPROVED** [review]
It looks like we could covert `findImportFor` into a `Decl::findImport(DeclContext *useDC)` as a follow-up.

**Comment by sophiapoirier** [comment]
@swift-ci please test

**Comment by sophiapoirier** [comment]
@swift-ci please test

**Pull request merged**